### PR TITLE
fix(ui5-segmented-button): fix visual design

### PR DESCRIPTION
**Pull request by hinzzx**
Segmented Button's visual design is now re-worked, by following the guidance of the VD Specification, ensuring pixel-perfect accuracy.

The integration of the new parameters are also addressed for RTL mode as well.

In the previous PR, we couldnt fully achieve the VD specification guidelines, because there were borders and box-shadows mixed and messed up which caused border overlapping, and small width shifts in the layouts sometimes. So In order to achieve the deisgn as it is intended we had to almost fully adopt using `box-shadow` instead of `border` for our component's visual representation.

The renaming of the files will be uploaded by separate PR, when this is merged.

### The guidelines:
image

### The result:
Snag_2027e309

Fixes: #7141

### CODE REVIEW

File: packages/main/src/themes/SegmentedButton.css [hunk 1]
```diff
@@ -28,17 +32,20 @@
...
+	border-radius: var(--_ui5_segmented_btn_item_hover_border_radius);
+	box-shadow: var(--_ui5_segmented_btn_item_hover_box_shadow);
 }
 
 ::slotted([ui5-segmented-button-item][pressed]),
 ::slotted([ui5-segmented-button-item][active]) {
-	border: 0.0625rem solid var(--sapButton_Selected_BorderColor);
+	box-shadow: var(--_ui5_segmented_btn_item_pressed_active_box_shadow);
+	border-radius: var(--_ui5_segmented_btn_item_pressed_active_border_radius);
 	background-color: var(--sapButton_Selected_Background);
```
Comment (unazko) [inline packages/main/src/themes/SegmentedButton.css:43]: Think it should be this one by spec `--sapButton_Active_Background`

File: packages/main/src/themes/SegmentedButton.css [hunk 2]
```diff
@@ -50,30 +57,60 @@
 
 ::slotted([ui5-segmented-button-item][pressed]:nth-child(odd)),
 ::slotted([ui5-segmented-button-item][active]:nth-child(odd)) {
-	border-inline-end: var(--_ui5_segmented_btn_inner_pressed_border_odd_child);
-	border-inline-start: var(--_ui5_segmented_btn_inner_pressed_border_odd_child);
+	box-shadow: var(--_ui5_segmented_btn_item_pressed_active_ODD_box_shadow);
```
Comment (unazko) [inline packages/main/src/themes/SegmentedButton.css:60]: --sapButton_Active_BorderColor parameter is not used, but it is present in the spec.
Instead there is the following parameter: _ui5_segmented_btn_item_pressed_active_box_shadow, which corresponds to --sapButton_Selected_BorderColor instead.

File: packages/main/src/themes/SegmentedButton.css (hunk 1 above)
Comment (unazko) [reply packages/main/src/themes/SegmentedButton.css:43]: That might need an override on for horizon only.

**Review by unazko — APPROVED** [review]

**Comment by hinzzx** [comment]
Closed due to another change #7528

**Pull request closed without merging**